src/mapping/Mapping.hpp

```diff
@@ -48,8 +51,23 @@ class Mapping {
     FULL = 2
   };
 
+  /**
+   * @brief Specifies whether the mapping is iterative or direct
+   *
+   * An iterative mapping uses an initial guess to perform the mapping.
```

Comment:
For completion, you could also describe direct mappings.

Is this classification referring to the mapping algorithms, or to their implementations in preCICE? Sounds a bit arbitrary, and I would need to look into each case to find out which is which. Adding some examples here would help.

My gut feeling tells me that this is a technical detail and that a simpler solution, without this classification, may be possible. Don't ask me what, though... :see_no_evil: